lib/Mojo/Log.pm

```diff
@@ -66,7 +67,20 @@ sub _default {
 sub _log {
   my ($self, $level) = (shift, pop);
   my @msgs = ref $_[0] eq 'CODE' ? $_[0]() : @_;
-  $msgs[0] = "$self->{context} $msgs[0]" if $self->{context};
+  if ($self->{context}) {
+    if (@msgs == 1 && blessed($msgs[0]) && $msgs[0]->isa('Mojo::Exception')) {
+      $msgs[0] = Mojo::Exception
+        ->new("$self->{context} ".$msgs[0]->message)
+        ->frames($msgs[0]->frames)
+        ->line($msgs[0]->line)
+        ->lines_after($msgs[0]->lines_after)
+        ->lines_before($msgs[0]->lines_before)
+        ->verbose($msgs[0]->verbose);
```

Comment:
Why so complicated?

Reply:
I don't know a simpler way to copy all the attributes of a `Mojo::Exception` (or `Mojo::Base`) object.